fuseimg/datasets/cmmd.py

```diff
@@ -0,0 +1,117 @@
+from fuse.data.pipelines.pipeline_default import PipelineDefault
+from fuse.data.datasets.dataset_default import DatasetDefault
+from fuse.data.datasets.caching.samples_cacher import SamplesCacher
+from fuseimg.data.ops.image_loader import OpLoadDicom
+from fuseimg.data.ops.color import OpNormalizeAgainstSelfImpl
+from fuseimg.data.ops.shape_ops import OpFlipBrightSideOnLeft2D , OpFindBiggestNonEmptyBbox2D, OpResizeAndPad2D
+from fuse.data import PipelineDefault, OpToTensor
+from fuse.data.ops.ops_common import OpLambda
+from fuseimg.data.ops.aug.color import OpAugColor
+from fuseimg.data.ops.aug.geometry import OpAugAffine2D 
+from fuse.data.ops.ops_aug_common import OpSample
+from fuse.data.ops.ops_read import OpReadDataframe
+from functools import partial
+import torch
+import pandas as pd
+from typing import Tuple, List
+import os
+
+from fuse.utils.rand.param_sampler import Uniform, RandInt, RandBool
+
+
+class CMMD:
+    """
+    # dataset that contains breast mamography biopsy info and metadata from chinese patients
+    # Path to the stored dataset location
+    # dataset should be download from https://wiki.cancerimagingarchive.net/pages/viewpage.action?pageId=70230508
+    # download requires NBIA data retriever https://wiki.cancerimagingarchive.net/display/NBIA/Downloading+TCIA+Images
+    # put on the folliwing in the main folder  - 
+    # 1. CMMD_clinicaldata_revision.csv which is a converted version of CMMD_clinicaldata_revision.xlsx 
+    # 2. folder named CMMD which is the downloaded data folder
+    """
+    # bump whenever the static pipeline modified
+    CMMD_DATASET_VER = 0
+
+    @staticmethod
+    def static_pipeline(data_dir: str,data_source : pd.DataFrame) -> PipelineDefault:
+        """
+        Get suggested static pipeline (which will be cached), typically loading the data plus design choices that we won't experiment with.
+        :param data_path: path to original kits21 data (can be downloaded by KITS21.download())
+        """
+        static_pipeline = PipelineDefault("static", [
+            
+            (OpReadDataframe(data_source,key_column = None , columns_to_extract = ['file','classification'] , rename_columns=dict(file="data.input.img_path",classification="data.gt.classification")), dict()), # will save image and seg path to "data.input.img_path", "data.gt.seg_path" 
+            (OpLoadDicom(data_dir), dict(key_in="data.input.img_path", key_out="data.input.img", format="nib")),
+            (OpFlipBrightSideOnLeft2D(), dict(key="data.input.img")),
+            (OpFindBiggestNonEmptyBbox2D(), dict(key="data.input.img")),
+            (OpNormalizeAgainstSelfImpl(), dict(key="data.input.img")),
```

Comment:
I know that it was already like this.
But can you please remove the "impl" suffix?

Reply:
no problem